lib/cartopy/mpl/gridliner.py

```diff
@@ -327,58 +338,257 @@ def _draw_gridliner(self, nx=None, ny=None, background_patch=None):
+        lon_lines = []
+        lat_min, lat_max = lat_lim
+        if lat_ticks:
+            lat_min = min(lat_min, min(lat_ticks))
+            lat_max = max(lat_max, max(lat_ticks))
+        for x in lon_ticks:
+            ticks = list(zip(
+                [x]*n_steps,
+                np.linspace(lat_min, lat_max, n_steps)))
+            lon_lines.append(ticks)
 
+        if self.xlines:
+            nx = len(lon_lines) + 1
+            # XXX this bit is cartopy specific. (for circular longitudes)
+            # Purpose: omit plotting the last x line,
+            # as it may overlap the first.
+            if (isinstance(crs, Projection) and
+                    isinstance(crs, _RectangularProjection) and
+                    abs(np.diff(lon_lim)) == abs(np.diff(crs.x_limits))):
+                nx -= 1
+            lon_lc = mcollections.LineCollection(lon_lines,
+                                                 **collection_kwargs)
+            self.xline_artists.append(lon_lc)
+            self.axes.add_collection(lon_lc, autolim=False)
+
+        # Parallels
+        lat_lines = []
+        lon_min, lon_max = lon_lim
+        if lon_ticks:
+            lon_min = min(lon_min, min(lon_ticks))
+            lon_max = max(lon_max, max(lon_ticks))
+        for y in lat_ticks:
+            ticks = list(zip(
+                np.linspace(lon_min, lon_max, n_steps),
+                [y]*n_steps))
+            lat_lines.append(ticks)
         if self.ylines:
-            lines = []
-            for y in y_ticks:
-                ticks = list(zip(
-                    np.linspace(min(x_ticks), max(x_ticks), n_steps),
-                    np.zeros(n_steps) + y))
-                lines.append(ticks)
-
-            y_lc = mcollections.LineCollection(lines, **collection_kwargs)
-            self.yline_artists.append(y_lc)
-            self.axes.add_collection(y_lc, autolim=False)
+            lat_lc = mcollections.LineCollection(lat_lines,
+                                                 **collection_kwargs)
+            self.yline_artists.append(lat_lc)
+            self.axes.add_collection(lat_lc, autolim=False)
 
         #################
         # Label drawing #
         #################
 
-        # Trim outside-area points from the label coords.
-        # Tickers may round *up* the desired range to something tidy, not
-        # all of which is necessarily visible.  We must be stricter with
-        # our texts, as they are drawn *without clipping*.
-        x_label_points = [x for x in x_ticks if x_lim[0] <= x <= x_lim[1]]
-        y_label_points = [y for y in y_ticks if y_lim[0] <= y <= y_lim[1]]
+        self.bottom_label_artists = []
+        self.top_label_artists = []
+        self.left_label_artists = []
+        self.right_label_artists = []
+        if not (self.left_labels or self.right_labels or
+                self.bottom_labels or self.top_labels):
+            return
+        self._assert_can_draw_ticks()
```

Comment:
This assertion is now incorrect here, is it not?

Reply:
In the sense that you can now draw labels on projections other than PlateCarree.